osu.Game/Rulesets/Ruleset.cs

```diff
@@ -158,7 +158,15 @@ protected Ruleset()
 
         public abstract DifficultyCalculator CreateDifficultyCalculator(WorkingBeatmap beatmap);
 
-        public virtual PerformanceCalculator CreatePerformanceCalculator(WorkingBeatmap beatmap, ScoreInfo score) => null;
+        public virtual PerformanceCalculator CreatePerformanceCalculator(DifficultyAttributes attributes, ScoreInfo score) => null;
+
+        [Obsolete("Use the DifficultyAttributes overload instead.")]
+        public PerformanceCalculator CreatePerformanceCalculator(WorkingBeatmap beatmap, ScoreInfo score)
```

Comment:
As stated above, I'd like more opinions on whether we should actually be obsoleting this.

Reply:
Could remain as non-obsolete - may be useful for osu!tools and similar usages. My bigger concern at this point is whether this is a reasonable breaking change to make, because I think the tau ruleset already has a pp calculator.

Reply:
Just did a symbol search on all available rulesets repositories and it seems no custom ruleset implemented a performance calculator yet (didn't got any results).

Reply:
I also did the same and saw no hits for overrides (but did see hits for calls of the method). I think the obsoletion warning should be removed on that basis (maybe we even want to make the overload that takes `DifficultyAttributes` `internal`? not sure)

Reply:
It's fine to keep the override public. And remove the obsoletion notice.